flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskManagerJobMetricGroup.java

```diff
@@ -80,8 +80,17 @@ public TaskMetricGroup addTask(
 					taskName,
 					subtaskIndex,
 					attemptNumber);
-				tasks.put(executionAttemptID, task);
-				return task;
+				TaskMetricGroup prior = tasks.put(executionAttemptID, task);
+				if (prior == null) {
+					return task;
```

Comment:
Can you avoid adding `closeLocally()` by simply doing a "contains()" check before creating the group?